### PR TITLE
Coverity Scan

### PR DESCRIPTION
**Pull request by WeiqunZhang**
## Summary

Fix issues found by Coverity Scan, including infinite loop bugs in the kernels of MLALaplacian.

The scan will be done nightly on a local workstation.

Add coverity scan and license badges to README.md. Remove the cmake README.md because it's broken.

## Checklist

The proposed changes:
- [x] fix a bug or incorrect behavior in AMReX
- [ ] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [ ] include documentation in the code and/or rst files, if appropriate

### CODE REVIEW

File: Tools/GNUMake/Make.defs [hunk 1]
```diff
@@ -991,6 +991,10 @@ ifeq ($(USE_CLANG_TIDY),TRUE)
   include        $(AMREX_HOME)/Tools/GNUMake/tools/Make.clang-tidy
 endif
 
+ifeq ($(USE_COVERITY),TRUE)
+  DEFINES += -DAMREX_USE_COVERITY
+endif
```
Comment (ax3l) [inline Tools/GNUMake/Make.defs:996]: We can add  this also to CMake, but if your build logic on Coverity is GNUmake only then this is fine, too.

**Review by ax3l — APPROVED** [review]
Looks great :+1:

**Comment by ax3l** [comment]
@WeiqunZhang just for sake of documentation, where did you implement the nightly sync-to-the-coverity-branch logic?

**Comment by WeiqunZhang** [comment]
I have a cron job running on this. We can only run 3 scans per day on average.
```
export PATH=$HOME/opt/cov-analysis/bin:$PATH

cd $HOME/coverity/amrex
git pull

rm -rf cov-int amrex.tgz

make distclean

./configure --with-mpi yes \
            --with-omp no \
            --with-fortran no \
            --enable-fortran-api no \
            --enable-eb yes \
            --enable-linear-solver yes \
            --enable-particle yes \
            --enable-tiny-profile yes

cov-build --dir cov-int make -j4 CXX=g++ USE_ASSERTION=TRUE USE_COVERITY=TRUE

tar cvfz amrex.tgz cov-int

AMREX_VERSION=$(git describe --abbrev=12 --dirty --always --tags)

curl --form token=${TOKEN} \
  --form email=weiqunzhang@lbl.gov \
  --form file=@amrex.tgz \
  --form version="AMReX-${AMREX_VERSION}" \
  --form description="AMReX" \
  https://scan.coverity.com/builds?project=AMReX-Codes%2Famrex
```

**Comment by ax3l** [comment]
Ah I see, got it.

In the past, I merged to a branch, which kicked off the upload via a plugin. Curling works as well.

If you like, you can also define cron-jobs in GH actions instead of locally:  https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule

**Comment by WeiqunZhang** [comment]
Good suggestion. We could do it.

**Comment by WeiqunZhang** [comment]
But then we have to download the coverity tools everytime.

**Comment by WeiqunZhang** [comment]
The tarball is quite big. More than 1 GB.

**Comment by ax3l** [comment]
Oh, I see. Yeah, it's on the size of the CUDA toolkit or oneAPI then :sweat_smile:

**Pull request merged**